### ISSUE TITLE
To simplify the creation of "semver" resources, trim leading/trailing whitespace before parsing the semver string

### ISSUE THREAD

**Issue by dylan-bourque** [concourse/semver-resource#30]
Currently, parsing the content of a semver file fails is there is trailing whitespace (\r, \n, \t, etc.) after the semantic version string.  I don't see any specifics about trailing whitespace in the [current semver spec](http://semver.org/spec/v2.0.0.html#semantic-versioning-specification-semver), so additional whitespace could be considered "invalid".

It would be much more "user-friendly", however, for the Concourse resource to trim extraneous whitespace from the contents of the semver file.  I say this because the most obvious end-user workflow (at least to me) creates an invalid resource.

```
echo 0.1.0 > myversion`
aws s3 cp myversion s3://mybucket/myversion

... create pipeline referencing the S3 bucket above as a "semver" resource

fly -t my-ci-server set-pipeline -p my-pipeline -c my-pipeline.yml
fly -t my-ci-server check-resource -r my-pipeline/my-semver-resource
```
The last command will return an error message like:
`error checking for new versions: parsing number in bucket: Invalid character(s) found in patch number "0\n"`

I was able to work around the error by using `echo -n` to generate the initial semver file, but that was definitely non-obvious.  I also tried creating the file using vi, vim, emacs, nano, and GEdit on my Ubuntu 16.04 system.  From what I saw, the default behavior of all of them was to include a trailing newline character in the file.

**Comment by tredfield** [comment]
@dylan-bourque thanks for the work around. we were beating our heads on our desks trying to figure out where the \n was coming from and how to resolve it.

**Comment by mbarzare** [comment]
@dylan-bourque I will echo what @tredfield said. Thanks so much identifying this problem and supplying a work around until it gets fixed!

**Comment by eedwardsdisco** [comment]
POSIX standard to end with a newline, github's code viewer even ignores the final newline for that reason. Definitely should be supported.

However, I've got 'POSIX newlines' in my version files, and it's working with concourse 2.7.0 and the Git driver. Possibly an S3 driver-specific problem?

**Comment by dylan-bourque** [comment]
@eedwardsdisco The error I'm seeing is coming from the call to `semver.Parse()` in `S3Driver.Check()` ([line 95 in .../driver/s3.go](https://github.com/concourse/semver-resource/blob/master/driver/s3.go#L95)).  Following that call into [the source](https://github.com/blang/semver/blob/master/semver.go#L280) shows that `Parse()` returns an error if the patch portion of the semver string contains anything other than digits, including any whitespace.

As I mentioned in my original description, the semver language spec is unclear about whether or not trailing whitespace is allowed.  That makes me lean towards stripping off the surrounding whitespace in the text pulled from the raw resource, S3 or otherwise, before calling `semver.Parse()` as the _best_ solution.

**Comment by jtarchie** [comment]
This is resolved in concourse/semver-resource#36
